ASan crash in `CEventManager::startThread` on FreeBSD

Regressed by a6cfe7042897. On [DragonFly](https://github.com/DragonFlyBSD/DragonFlyBSD/blob/DragonFly_RELEASE_6_4/sys/sys/un.h#L49), [FreeBSD](https://github.com/freebsd/freebsd-src/blob/releng/13.2/sys/sys/un.h#L52), [NetBSD](https://github.com/NetBSD/src/blob/netbsd-9/sys/sys/un.h#L52), [OpenBSD](https://github.com/openbsd/src/blob/be82246fa6d2ed41b36408de60ad9512878fa3a9/sys/sys/un.h#L52) `sun_path` size is `104`.

```c++
$ meson setup --buildtype=debug -Db_sanitize=address /tmp/hyprland_build
$ meson compile -C /tmp/hyprland_build -c /dev/null
$ /tmp/hyprland_build/src/Hyprland -c /dev/null
[...]
00:00:00.040 [backend/headless/backend.c:68] Creating headless backend
=================================================================
==85957==ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7fffddfedc2a at pc 0x00000068d48c bp 0x7fffddfedb90 sp 0x7fffddfed348
WRITE of size 107 at 0x7fffddfedc2a thread T17
00:00:00.055 [wayland] unable to lock lockfile /var/run/user/1001/wayland-1.lock, maybe another compositor is running
00:00:00.055 [backend/wayland/backend.c:427] Starting Wayland backend
    #0 0x68d48b in strncpy (/tmp/hyprland_build/src/Hyprland+0x68d48b)
    #1 0x92ebaa in CEventManager::startThread()::$_0::operator()() const EventManager.cpp
    #2 0x92e7b4 in decltype(std::declval<CEventManager::startThread()::$_0>()()) std::__1::__invoke[abi:v160002]<CEventManager::startThread()::$_0>(CEventManager::startThread()::$_0&&) EventManager.cpp
    #3 0x92e78c in void std::__1::__thread_execute[abi:v160002]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, CEventManager::startThread()::$_0>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, CEventManager::startThread()::$_0>&, std::__1::__tuple_indices<>) EventManager.cpp
    #4 0x92e41b in void* std::__1::__thread_proxy[abi:v160002]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, CEventManager::startThread()::$_0>>(void*) EventManager.cpp
    #5 0x8017beb74 in thread_start /usr/src/lib/libthr/thread/thr_create.c:292:16

Address 0x7fffddfedc2a is located in stack of thread T17 at offset 138 in frame
    #0 0x92e7ef in CEventManager::startThread()::$_0::operator()() const EventManager.cpp

  This frame has 8 object(s):
    [32, 138) 'SERVERADDRESS'
    [176, 200) 'socketPath' <== Memory access at offset 138 partially underflows this variable
    [240, 264) 'ref.tmp' <== Memory access at offset 138 partially underflows this variable
    [304, 320) 'clientAddress'
    [336, 340) 'clientSize'
    [352, 356) 'ACCEPTEDCONNECTION'
    [368, 384) 'ref.tmp28'
    [400, 408) 'ref.tmp29'
HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
      (longjmp and C++ exceptions *are* supported)
Thread T17 created by T0 here:
    #0 0x68859a in pthread_create (/tmp/hyprland_build/src/Hyprland+0x68859a)
    #1 0x93eb78 in std::__1::__libcpp_thread_create[abi:v160002](pthread**, void* (*)(void*), void*) EventManager.cpp
    #2 0x92cf88 in std::__1::thread::thread<CEventManager::startThread()::$_0, void>(CEventManager::startThread()::$_0&&) EventManager.cpp
    #3 0x92cd86 in CEventManager::startThread() (/tmp/hyprland_build/src/Hyprland+0x92cd86)
    #4 0xc523fe in CCompositor::initManagers(eManagersInitStage) (/tmp/hyprland_build/src/Hyprland+0xc523fe)
    #5 0xc51bc4 in CCompositor::initServer() (/tmp/hyprland_build/src/Hyprland+0xc51bc4)
    #6 0xe0c483 in main (/tmp/hyprland_build/src/Hyprland+0xe0c483)
    #7 0x80187d189 in __libc_start1 /usr/src/lib/libc/csu/libc_start1.c:155:7
    #8 0x572541 in _start /usr/src/lib/csu/amd64/crt1_c.c:52:2
    #9 0x801065007  (<unknown module>)

SUMMARY: AddressSanitizer: stack-buffer-overflow (/tmp/hyprland_build/src/Hyprland+0x68d48b) in strncpy
Shadow bytes around the buggy address:
  0x7fffddfed980: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7fffddfeda00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7fffddfeda80: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7fffddfedb00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7fffddfedb80: 00 00 00 00 f1 f1 f1 f1 00 00 00 00 00 00 00 00
=>0x7fffddfedc00: 00 00 00 00 00[02]f2 f2 f2 f2 00 00 00 f2 f2 f2
  0x7fffddfedc80: f2 f2 f8 f8 f8 f2 f2 f2 f2 f2 f8 f8 f2 f2 f8 f2
  0x7fffddfedd00: f8 f2 f8 f8 f2 f2 f8 f3 f3 f3 f3 f3 00 00 00 00
  0x7fffddfedd80: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7fffddfede00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7fffddfede80: 00 00 00 00 00 00 00 00 00 00 00 00 f1 f1 f1 f1
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==85957==ABORTING
```

This also shows up during build but drowns in the noise of other warnings:
```c++
hyprctl/main.cpp:83:5: warning: 'strncpy' size argument is too large; destination buffer has size 104, but size argument is 107 [-Wfortify-source]
    strncpy(serverAddress.sun_path, socketPath.c_str(), 107);
    ^
hyprctl/main.cpp:146:5: warning: 'strncpy' size argument is too large; destination buffer has size 104, but size argument is 107 [-Wfortify-source]
    strncpy(serverAddress.sun_path, socketPath.c_str(), 107);
    ^
src/managers/EventManager.cpp:70:9: warning: 'strncpy' size argument is too large; destination buffer has size 104, but size argument is 107 [-Wfortify-source]
        strncpy(SERVERADDRESS.sun_path, socketPath.c_str(), 107);
        ^
```

--- a/hyprctl/main.cpp
+++ b/hyprctl/main.cpp
@@ -80,7 +80,7 @@ void              request(std::string arg, int minArgs = 0) {
 
     std::string socketPath = "/tmp/hypr/" + instanceSigStr + "/.socket.sock";
 
-    strncpy(serverAddress.sun_path, socketPath.c_str(), 107);
+    strncpy(serverAddress.sun_path, socketPath.c_str(), sizeof(serverAddress.sun_path) - 1);
 
     if (connect(SERVERSOCKET, (sockaddr*)&serverAddress, SUN_LEN(&serverAddress)) < 0) {
         std::cout << "Couldn't connect to " << socketPath << ". (3)";
@@ -143,7 +143,7 @@ void requestHyprpaper(std::string arg) {
 
     std::string socketPath = "/tmp/hypr/" + instanceSigStr + "/.hyprpaper.sock";
 
-    strncpy(serverAddress.sun_path, socketPath.c_str(), 107);
+    strncpy(serverAddress.sun_path, socketPath.c_str(), sizeof(serverAddress.sun_path) - 1);
 
     if (connect(SERVERSOCKET, (sockaddr*)&serverAddress, SUN_LEN(&serverAddress)) < 0) {
         std::cout << "Couldn't connect to " << socketPath << ". (3)";

--- a/src/managers/EventManager.cpp
+++ b/src/managers/EventManager.cpp
@@ -67,7 +67,7 @@ void CEventManager::startThread() {
 
         sockaddr_un SERVERADDRESS = {.sun_family = AF_UNIX};
         std::string socketPath    = "/tmp/hypr/" + g_pCompositor->m_szInstanceSignature + "/.socket2.sock";
-        strncpy(SERVERADDRESS.sun_path, socketPath.c_str(), 107);
+        strncpy(SERVERADDRESS.sun_path, socketPath.c_str(), sizeof(SERVERADDRESS.sun_path) - 1);
 
         bind(SOCKET, (sockaddr*)&SERVERADDRESS, SUN_LEN(&SERVERADDRESS));
 

